src/Infrastructure.Styles.Analyzer/RMSYN1102SeparateExitConditionsAnalyzer/IfAndElseIfStatementSyntaxWrapper.cs

```diff
@@ -0,0 +1,134 @@
+// This file is part of the re-motion Framework (www.re-motion.org)
+// Copyright (c) rubicon IT GmbH, www.rubicon.eu
+// 
+// re-motion Framework is free software; you can redistribute it
+// and/or modify it under the terms of the GNU Lesser General Public License
+// as published by the Free Software Foundation; either version 2.1 of the
+// License, or (at your option) any later version.
+// 
+// re-motion is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Lesser General Public License for more details.
+// 
+// You should have received a copy of the GNU Lesser General Public License
+// along with re-motion; if not, see http://www.gnu.org/licenses.
+
+using System;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+
+namespace Infrastructure.Styles.Analyzer.RMSYN1102SeparateExitConditionsAnalyzer
+{
+  public class IfAndElseIfStatementSyntaxWrapper
+  {
+    public SyntaxNode IfOrElseIfStatement { get; private set; }
+    public ElseClauseSyntax? ElseClause { get; }
+    public bool HasElseClause { get; }
+
+    public IfAndElseIfStatementSyntaxWrapper (SyntaxNode ifOrElseIfStatement, SyntaxNode? elseClause)
+    {
+      if (ifOrElseIfStatement is not IfStatementSyntax and not ElseClauseSyntax)
+        throw new ArgumentException("SyntaxNode must be of kind IfStatementSyntax or ElseClauseSyntax");
+      IfOrElseIfStatement = ifOrElseIfStatement;
+      if (elseClause != null && elseClause.IsKind(SyntaxKind.ElseClause))
+      {
+        ElseClause = (ElseClauseSyntax) elseClause;
+        HasElseClause = true;
+      }
+      else
+      {
+        HasElseClause = false;
+      }
```

Comment:
Doesn't work, im not quite sure why, but it doesnt. Weird.

Reply:
Nevermind, I think the problem is that as doesn't care if the input is of the actual type, it just returns null otherwise and we checked that in the first iteration.

Reply:
`HasElseClause = ElseClause != null` is missing, by bad :D